diff --git a/src/abort_message.cpp b/src/abort_message.cpp
index 859a503..b761a0d 100644
--- a/src/abort_message.cpp
+++ b/src/abort_message.cpp
@@ -11,6 +11,10 @@
 #include <stdarg.h>
 #include "abort_message.h"
 
+#if defined(__MUSL_OHOS__)
+extern "C" void set_fatal_message(const char *msg);
+#endif  // defined(__MUSL_OHOS__)
+
 #ifdef __BIONIC__
 #   include <android/api-level.h>
 #   if __ANDROID_API__ >= 21
@@ -42,6 +46,11 @@ void abort_message(const char* format, ...)
     }
 #endif
 
+#if defined(__MUSL_OHOS__)
+    if (&set_fatal_message != nullptr) {
+        set_fatal_message(format);
+    }
+#endif  // defined(__MUSL_OHOS__)
     // Format the arguments into an allocated buffer. We leak the buffer on
     // purpose, since we're about to abort() anyway.
 #if defined(_LIBCXXABI_USE_CRASHREPORTER_CLIENT)
-- 